cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -262,6 +262,36 @@ fn emit_std_enc_mem(
                 panic!("ImmRegRegShift");
             }
         }
+
+        Amode::RipRelative { ref target } => {
+            // First, the REX byte, with REX.B = 0.
+            rex.emit_two_op(sink, enc_g, 0);
+
+            // Now the opcode(s).  These include any other prefixes the caller
+            // hands to us.
+            while num_opcodes > 0 {
+                num_opcodes -= 1;
+                sink.put1(((opcodes >> (num_opcodes << 3)) & 0xFF) as u8);
+            }
+
+            // RIP-relative is mod=00, rm=101.
+            sink.put1(encode_modrm(0, enc_g & 7, 0b101));
+
+            match *target {
+                BranchTarget::Label(label) => {
+                    let offset = sink.cur_offset();
+                    sink.use_label_at_offset(offset, label, LabelUse::JmpRel32);
```

Comment:
Interestingly, there's also a `PCRel32` in addition to `JmpRel32`: the former adds the value read at the offset before patching, while the latter does subtract 4 to the read value before patching (because control flow instructions are relative to the start of the next instruction on x64). This is one of these cases here. Should i rename `JmpRel32` to `PCRel32Disp4` or something like this instead?

Reply:
Maybe `LabelUse`s need to be architecture-specific, then, if they are not already?

Reply:
They are architecture-specific, already: this is defined in inst/mod.rs.